consensus/wal_test.go

```diff
@@ -32,17 +34,23 @@ func TestWALTruncate(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	wal.group.SetHeadSizeLimit(4096) //cause fragment
+	//this magic number 4K can truncate the content when RotateFile. defaultHeadSizeLimit(10M) is hard to simulate.
+	wal.Group().SetHeadSizeLimit(4096)
```

Comment:
same here. we shouldn't expose SetXXX methods for something which is not supposed to change during the WAL group lifetime.

Reply:
We would appreciate if you refactor this part! Technical debt tends to accumulate over time, so we need to refactor to maintain Tendermint in good shape continually.